### PR TITLE
Replace n1 highcpu 8 with e2 highcpu 8

### LINKED ISSUE #3486: Switch couldbuilds from N1 to E2 Machines
This issue is to apply https://github.com/kubernetes/k8s.io/issues/5059

In https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/cloudbuild.yaml, `machineType: 'N1_HIGHCPU_8'` should be replaced with `machineType: E2_HIGHCPU_32` 

/good-first-issue

### PR DESCRIPTION
**Pull request by sepehrsoh**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3486

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by mboersma — COMMENTED** [review]
/lgtm
/assign @CecileRobertMichon

File: cloudbuild.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 timeout: 2700s
 options:
   substitution_option: ALLOW_LOOSE
-  machineType: 'N1_HIGHCPU_8'
+  machineType: 'E2_HIGHCPU_32'
```
[the remark this answers is not in the document]
Comment (sepehrsoh) [reply cloudbuild.yaml]: @CecileRobertMichon Done

[the remark this answers is not in the document]
Comment (CecileRobertMichon) [reply cloudbuild.yaml]: thank you 🙏

**Comment by CecileRobertMichon** [comment]
@sepehrsoh can you please squash the commits? Thanks!

/lgtm

**Comment by sepehrsoh** [comment]
> @sepehrsoh can you please squash the commits? Thanks!
> 
> /lgtm

sure! done

**Review by mboersma — APPROVED** [review]
/lgtm
/approve

**Pull request merged**